conjur-api/Client.Methods.cs

```diff
@@ -24,6 +28,33 @@ public Variable Variable(string name)
             return new Variable(this, name);
         }
 
+        /// <summary>
+        /// Search for variables
+        /// </summary>
+        /// <param name="query">Query for search.</param>
+        /// <param name="actingAs">Fully-qualified Conjur ID of a role to act as.</param>
+        /// <returns>Returns IEnumerable to VariableInfo.</returns>
+        public IEnumerable<Variable> SearchVariables(string query = null, string actingAs = null)
```

Comment:
This file is a facade file; the body of this method should go elsewhere.

Also, there is nothing specific to variables about this method; this is just a generic resource listing. It should go to Resource class.

(The name is my fault @orenbm , I suggested it. In retrospect, it should be `static IEnumerable<Resource> Resource.List(Client client, string kind, string search = null)` and this facade should have a `IEnumerable<Resource> ListResources(string kind, string search = null)` proxy method.)

Reply:
Also, I'll note the signature of this method is incongruent with the docstring.

Reply:
I think it's clearer to have this method in Client.Methods as we need some info from the client (i.e Account) and it makes more sense to have the client retrieve resources.

I agree that we should have a method for ListResources and have most of the logic there but as a user I'd like to have these methods:

client.ListVariables()
client.ListUsers()
etc.

These methods should call ListResources(enum kind) which will do most of the logic.

WDYT?